src/uu/head/src/head.rs

```diff
@@ -431,13 +433,37 @@ fn uu_head(options: &HeadOptions) -> UResult<()> {
                 match options.mode {
                     Modes::Bytes(n) => {
                         if options.all_but_last {
+                            // Outputting all but last requires us to use a ring buffer with size n,
+                            // so n must be converted from u64 to usize to fit in memory.
+                            let n: usize = match n.try_into() {
+                                Ok(n) => n,
+                                Err(n) => {
+                                    show!(USimpleError::new(
+                                        1,
+                                        format!("{}: number of bytes is too large", n)
+                                    ));
+                                    continue;
+                                }
+                            };
```

Comment:
And this may conflict with #2906.

Reply:
Thanks for the heads-up, I will politely wait for that to merge and fix whatever conflicts will arise